orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java

```diff
@@ -110,8 +111,25 @@ private void addTriggers(
       pipeline.put(
           "triggers",
           TemplateMerge.mergeDistinct(
-              (List<HashMap<String, Object>>) template.getPipeline().get("triggers"),
+              markInherited(getTemplateCollection(template, "triggers")),
               configuration.getTriggers()));
     }
   }
+
+  private static List<HashMap<String, Object>> markInherited(
+      List<HashMap<String, Object>> templateCollection) {
+    return templateCollection.stream()
+        .map(
+            templateItem -> {
+              templateItem.put("inherited", true);
+              return templateItem;
+            })
+        .collect(Collectors.toList());
+  }
+
+  private static List<HashMap<String, Object>> getTemplateCollection(
```

Comment:
```suggestion
  private static List<Map<String, Object>> getTemplateCollection(
```

Reply:
Ah good call out, definitely, thanks!

Reply:
hmmm now I recall why I did this, configuration.getParameters returns a HashMap and mergeDistinct expects both arguments to be of the same type

Reply:
ooooh gotcha :+1: